native/src/components/OpeningEntry.tsx

```diff
@@ -1,17 +1,93 @@
-import React, { ReactElement } from 'react'
-import { useTranslation } from 'react-i18next'
+import React, { ReactElement, useState } from 'react'
+import { Trans, useTranslation } from 'react-i18next'
+import { Modal as RNModal, Text } from 'react-native'
 import styled from 'styled-components/native'
 
 import { TimeSlot } from 'shared/api/types'
 
+import { NoteIcon } from '../assets'
 import { contentDirection } from '../constants/contentDirection'
+import Link from './Link'
+import Icon from './base/Icon'
+
+const BackgroundForClosing = styled.Pressable`
+  flex: 1;
+  background-color: ${props => props.theme.colors.textDecorationColor};
+  opacity: 0.25;
+`
+
+const OverlayCenterer = styled.View`
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  justify-content: center;
+  align-items: center;
+`
+
+const OverlayContainer = styled.View`
+  background-color: white;
+  margin: 32px;
+  padding: 24px;
+  border-radius: 28px;
+`
+
+const OverlayTitle = styled.Text`
+  font-weight: 700;
+  margin-bottom: 16px;
+`
+
+const CloseButton = styled.Pressable`
+  padding: 12px;
+  margin-left: auto;
+`
+
+const CloseButtonText = styled.Text`
+  font-weight: 700;
+  color: ${props => props.theme.colors.textSecondaryColor};
+`
+
+type AppointmentOnlyOverlayProps = {
+  link: string | null
+  closeOverlay: () => void
+}
+
+const AppointmentOnlyOverlay = ({ link, closeOverlay }: AppointmentOnlyOverlayProps): ReactElement => {
+  const { t } = useTranslation('pois')
+  return (
+    <RNModal onRequestClose={closeOverlay} animationType='fade' transparent>
```

Comment:
Our custom modal is made to fill in the entire screen, has a header with a close button, and I think there was another reason for why it didn't quite work without heavy re-writing of our modal component so I decided against it.

If we encounter another place where we need a smaller overlay like this, we could export it into a shared component then, but I think it would be a bit of a premature optimisation at the moment.

Reply:
Thanks for explaining!